doc/source/user_guide/local-usage.rst

```diff
@@ -0,0 +1,95 @@
+.. _local_usage:
+
+.. currentmodule:: nomenclature
+
+Local usage of a project
+========================
+
+.. attention:: This page is intended for users who are familiar with Python and
+    `git <https://git-scm.com>`_ (or a service like `GitHub <https://github.com>`_).
+
+You can use the **nomenclature** package locally (on your machine) for validation
+and region-aggregration. This can be helpful as part of processing your model results,
+or to ensure that a submission to an IIASA Scenario Explorer instance will succeed.
+
+Requirements
+------------
+
+1. Install the **nomenclature** package (see the :ref:`installation` instructions)
+2. Have a project folder that has the required :ref:`dir_structure`:
+
+   * Set up the folder with the required structure and files yourself, or
+   * Clone the git repository for a specific project
+     (e.g., `openENTRANCE <https://github.com/openENTRANCE/openentrance>`_)
+
+   .. attention:: When using a project repository from GitHub or a similar service,
+          make sure that you keep your local clone in sync with the upstream repository.
+
+Usage options
+-------------
+
+Validation against the codelists
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The easiest use case is to validate that a data file or an :class:`IamDataFrame <pyam.IamDataFrame>`
+is compatible with the codelists (lists of variables and regions) of a project's
+:class:`DataStructureDefinition`.
+
+If the :func:`process` function returns an :class:`IamDataFrame <pyam.IamDataFrame>`,
```

Comment:
Since this example is talking about validation only maybe we should stress that the dataframe is returned unchanged.
```suggestion
If there are any inconsistencies with the codelists an error will be raised, otherwise the :class:`IamDataFrame <pyam.IamDataFrame>` is returned unchanged.
```
or something along those lines.